nixos/modules/services/web-apps/libretranslate.nix

```diff
@@ -0,0 +1,203 @@
+      dataDir = lib.mkOption {
+        description = lib.mdDoc "The data directory.";
+        type = lib.types.path;
+        default = "/var/lib/libretranslate";
+        example = "/opt/libretranslate";
+      };
+      threads = lib.mkOption {
+        description = lib.mdDoc "Set number of threads.";
+        type = lib.types.nullOr lib.types.int;
+        default = null;
+        example = 8;
+      };
+      enableApiKeys = lib.mkOption {
+        description = lib.mdDoc "Enable API keys database.";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      disableWebUI = lib.mkOption {
+        description = lib.mdDoc "Disable web ui.";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      updateModels = lib.mkOption {
+        description = lib.mdDoc "Update language models at startup";
+        type = lib.types.bool;
+        default = false;
+        example = true;
+      };
+      domain = lib.mkOption {
+        description = lib.mdDoc "The domain serving your LibreTranslate instance. Required for configure nginx as a reverse proxy.";
+        type = lib.types.str;
+        default = "";
+        example = "libretranslate.example.com";
+      };
+      configureNginx = lib.mkOption {
+        description = lib.mdDoc "Configure nginx as a reverse proxy for LibreTranslate.";
+        type = lib.types.bool;
+        default = false;
+      };
+      extraArgs = lib.mkOption {
+        description = lib.mdDoc "Extra arguments passed to the LibreTranslate.";
+        type = lib.types.listOf lib.types.str;
+        default = [ ];
+        example = [
+          "--debug"
+          "--disable-files-translation"
+        ];
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ ltmanageKeysCli ];
+
+    systemd.tmpfiles.rules = lib.mkIf (cfg.dataDir != "/var/lib/libretranslate") [
+      "d '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+      "z '${cfg.dataDir}' 0750 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.libretranslate = {
+      description = "LibreTranslate service";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        HOME = cfg.dataDir;
+      };
+      serviceConfig = lib.mkMerge [{
+        Type = "simple";
+        ExecStart = ''
+          ${cfg.package}/bin/libretranslate \
+            --host ${cfg.host} \
+            --port ${toString cfg.port} \
+            ${lib.optionalString cfg.enableApiKeys "--api-keys"} \
+            ${lib.optionalString cfg.disableWebUI "--disable-web-ui"} \
+            ${lib.optionalString cfg.updateModels "--update-models"} \
+            ${lib.optionalString (cfg.threads != null) "--threads ${toString cfg.threads}"} \
+            ${lib.concatStringsSep " " cfg.extraArgs}
+        '';
+        WorkingDirectory = cfg.dataDir;
+        # User and group
+        User = cfg.user;
```

Comment:
Why not use [DynamicUser](https://github.com/NixOS/rfcs/blob/master/rfcs/0052-dynamic-ids.md)?

Reply:
DynamicUser is a more advanced feature which can be complicated to use. I won't be blocking PRs if people are not using it.

Reply:
I prefer to use the default option of adding an individual user. Allows to specify a user that is already in system, if necessary.